### PR TITLE
Configuration

### PR DESCRIPTION
**Pull request by broad-well**
Please take a look at this proposed implementation for configuration. Configuration support, as discussed in its Project description, is meant for the retrieval of volatile parameters from the phone's filesystem. This practice saves a significant amount of turnaround time and consequently allows for greater efficiency.

More documentation might be necessary, especially for the "standard directory for configuration files" reference in `Configuration.fromPropertiesFile`.

### CODE REVIEW

**Comment by broad-well** [comment]
I think it'd be a good idea to incorporate some examples in the TeamCode module (just as FTC does in FtcRobotController).

Also, Codacy doesn't like the inline initialization that I included for `schemaProvider` in `ConfigurationTest`. The alternatives for Java 7 are Guava and Kotlin. Pick your poison or ignore Codacy.

**Comment by jzhu2020** [comment]
I don't think I understand the purpose of the changes you have been making.

If you want me to review your changes, could you explain what they are trying to do?

**Comment by broad-well** [comment]
@jzhu2020 Are you referring to the last commit or this PR?

**Comment by broad-well** [comment]
This is the proposed implementation for the Configuration Project, which has the following description:

_Parameters that require frequent tweaks during the season should be defined in configuration files for quick turnaround times. (You don't want to recompile the entire app just to change a drivetrain ratio, correct?)_

**Comment by jzhu2020** [comment]
Thanks, that's more clear now.

However, I don't think I understand how you did it.

**Comment by broad-well** [comment]
Have you taken a look at the source? If the documentation is not clear to you, feel free to comment directly on them. (Hover over the line and you'll find a blue "plus" button next to the line number)

**Comment by jzhu2020** [comment]
I'll try commenting on anything I don't understand then.

**Comment by broad-well** [comment]
You are reviewing the code before without JavaDocs. Check out the "Files Changed" tab for this PR.

File: ARCCore/src/main/com/andoverrobotics/core/config/Configuration.java [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+import java.io.Reader;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+
+/**
+ * A key-value mapping abstraction that allows for the retrieval of values in a variety of types.
+ * <p> <h2>Schemas</h2> Configuration Schemas are small classes that only contain non-private fields
```
Comment (broad-well) [inline ARCCore/src/main/com/andoverrobotics/core/config/Configuration.java]: @jzhu2020 Here is the documentation you were looking for.

Comment (jzhu2020) [reply ARCCore/src/main/com/andoverrobotics/core/config/Configuration.java]: Alright, thanks.

**Pull request merged**